docs/language/contracts.rst

```diff
@@ -18,18 +18,21 @@ like so:
       }
   }
 
-A constructor does not have a name and may have any number of arguments. If a constructor has arguments,
-then when the contract is deployed then those arguments must be supplied.
+A constructor can have any number of arguments.
+If a constructor has arguments, those arguments must be supplied when the contract is deployed.
 
 If a contract is expected to hold receive value on instantiation, the constructor should be declared ``payable``.
 
 .. note::
+    In Substrate, constructors have a name. Solang allows to name constructors in the substrate target:
 
-  Parity Substrate allows multiple constructors to be defined, which is not true for
-  Solana. So, when building for Substrate, multiple constructors can be
-  defined as long as their argument list is different (i.e. overloaded).
+    .. code-block:: solidity
+ 
+        contract Foo {
+            constructor my_new_foo() {}
+        }
 
-  When the contract is deployed in the Polkadot UI, the user can select the constructor to be used.
+    Unnamed constructors will be called ``new`` in the metadata.
```

Comment:
I think we should mention that the constructor name is only used for the metadata and cannot be used for with the `new contract()` syntax.